var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -47,6 +48,7 @@ class Netcdf(Package):
 
     # Required for NetCDF-4 support
     depends_on("zlib")
+    depends_on("hdf5@:1.8.999", when='+hdf5_18')
```

Comment:
It seems that having three separate `depends_on` statements doesn't work since Spack doesn't combine them properly. The error is:
```
$ spack spec netcdf
Input spec
------------------------------
  netcdf

Normalized
------------------------------
  netcdf
      ^curl
          ^openssl
              ^zlib
      ^hdf5@:1.8
      ^m4

Concretized
------------------------------
==> Error: Invalid spec: 'hdf5@1.8.16%gcc@6.1.0-spack+cxx~debug+fortran~mpi+shared~szip~threadsafe arch=darwin-elcapitan-x86_64'. Package hdf5 requires variant +mpi, but spec asked for ~mpi
```

These two specifications work:
```
    depends_on("hdf5@:1.8+mpi", when='+mpi')
    depends_on("hdf5@:1.8~mpi", when='~mpi')
```
I will update the PR.